pkg/service/service.go

```diff
@@ -1466,43 +1473,38 @@ func (s *Service) updateBackendsCacheLocked(svc *svcInfo, backends []lb.Backend)
 						backend.L3n4Addr, err)
 				}
 				backends[i].ID = id
-				// Default backend state is active.
-				backends[i].State = lb.BackendStateActive
+				backends[i].Weight = backend.Weight
 				newBackends = append(newBackends, backends[i])
 				// TODO make backendByHash by value not by ref
 				s.backendByHash[hash] = &backends[i]
 			} else {
 				backends[i].ID = s.backendByHash[hash].ID
-				backends[i].State = s.backendByHash[hash].State
 			}
 			svc.backendByHash[hash] = &backends[i]
 		} else {
 			backends[i].ID = b.ID
-			// Update backend state.
-			if b.RestoredFromDatapath {
-				backends[i].State = b.State
-				// Toggle the flag as the backend is now restored.
-				b.RestoredFromDatapath = false
```

Comment:
Can you explain why you removed this code path? Was this intentional?

Reply:
Yeah this was intentional, so let's say I:
1) Add a service with some number of backends - all backend weight 1. All is good (all state active).
2) I restart the cilium-agent.
3) Now I want to put one of the backends out for maintenance. I run:
```
cilium service update --id 1 --frontend "10.248.11.13:7047" --backends "10.246.3.34:7047,10.246.39.33:7047" --k8s-node-port --backend-weights 1,0
```
But if I keep this code as is, the backend at this stage has flag `RestoredFromDatapath` set to true, which means the backend state is replaced here by the previous state (the one before restart, which was active) and this means I have to run this (the `cilium service update`...) twice to actually change the backend state.

*) with the changes I did:
I am able to run this only once and there is no problem AFAICS.

Reply:
This breaks the existing logic where we restore backend states from BPF maps. If you notice the code path (runtime CI test should fail too), once the backends are restored after agent restart, the `RestoredFromDatapath` flag is toggled. You would need to re-apply backend weights once all the service backends are [restored](http://github.com/cilium/cilium/blob/master/pkg/service/service.go#L822-L822).

Reply:
Why do you think so? The flag `RestoredFromDatapath` is se to true for each backend when agent starts up (that's true). But while backends are restored only function `NewBackendWithState() is executed, nothing else. Inside that function the state is correctly recovered from the bpf maps as you know. But that's it and I am not changing anything there. 

The runtime tests passed just fine without any hiccup. 
```
Ran 67 of 414 Specs in 3751.755 seconds
SUCCESS! -- 67 Passed | 0 Failed | 9 Pending | 338 Skipped

Test Suite Passed
```
.. I ran `make test` from test directory.

Reply:
and this function `updateBackendsCacheLocked` where I removed the code path (the one we are talking about) is never executed while restoration so there can't be any issue (related to my changes) while the backend is restored.